rasa/core/config.py

```diff
@@ -27,3 +28,45 @@ def load(config_file: Optional[Union[Text, Dict]]) -> List["Policy"]:
         config_data = config_file
 
     return PolicyEnsemble.from_dict(config_data)
+
+
+def migrate_mapping_policy_to_rules(
+    config: Dict[Text, Any], domain: "Domain", rules: List[Dict[Text, Any]]
+):
+    """
+    Migrate MappingPolicy to the new RulePolicy,
+    by updating the config, domain and generating rules.
+
+    This function modifies the config, the domain and the rules in place.
+    """
+    policies = config.get("policies", [])
+    has_mapping_policy = False
+    has_rule_policy = False
+
+    for policy in policies:
+        if policy.get("name") == "MappingPolicy":
+            has_mapping_policy = True
+        if policy.get("name") == "RulePolicy":
+            has_rule_policy = True
+
+    if not has_mapping_policy:
+        return
+
+    has_one_triggered_action = False
+    for intent, properties in domain.intent_properties.items():
+        # remove triggers from intents, if any
+        triggered_action = properties.pop("triggers", None)
+        if triggered_action:
+            has_one_triggered_action = True
+            rules.append(
+                {
+                    "rule": f"Rule to map `{intent}` intent (automatic conversion)",
```

Comment:
I'm currently at a similar place for the fallback rules: Using `DialogueStateTracker.from_events` feels more robust as the training data keys / format might change. Using pure yaml somewhat mixes the knowledge of the training data format into this module.

Reply:
Or using `YAMLStoryReader.read_from_string` seems a good middle way as it validates the format

Reply:
I'm not super familiar with the tracker. If you have an example to share that'd be simpler for me to understand 😄

Reply:
That's how I solved it now: https://github.com/RasaHQ/rasa/pull/6862/files#diff-ec2e0e4e9cc6048d7680bf9ccfe289beR115-R125 wdyt?

Reply:
I don't think it's ideal but it's at least an improvement with the schema validation. It would be rad if we had data classes on which we could rely, and which the rules would be automatically loaded into

Reply:
We have data classes (`DialogueStateTracker`, `StoryStep`). Maybe their initializers are not ideal. I think defining the rule in yaml, parsing it and then adding is a good compromise

Reply:
but there is no "Rule" and "RuleStep" dataclass, right? (and by data class, I mean https://docs.python.org/3/library/dataclasses.html)

Reply:
I'll let you consolidate this in your PR then